guides/messaging-high-availability.adoc

```diff
@@ -0,0 +1,227 @@
+--
+
+[[prerequisites]]
+== Prerequisites
+
+To complete this guide, you need:
+
+* Roughly {prerequisites-time} minutes
+* JDK {jdk-minimal-version} installed with `JAVA_HOME` configured appropriately
+
+== Prepare shared journal directories
+
+Create two directories for each primary-secondary pair:
+
+[source, bash ,options="nowrap"]
+----
+mkdir messaging-journal-a
+mkdir messaging-journal-b
+----
+
+== Prepare WildFly Servers
+
+Now let's configure two WildFly servers with two primary-secondary pairs of embedded ActiveMQ Artemis brokers. ActiveMQ Artemis configured
+as primary on the first WildFly server and as secondary on the second WildFly server will use `messaging-journal-a`
+directory for their shared journal. Similarly primary broker on second WildFly server and secondary broker on
+the first WildFly server will use `messaging-journal-b` directory for their shared journal.
+
+[NOTE]
+--
+All ActiveMQ Artemis brokers must be in cluster so they know about each other and can provide cluster topology to connected clients. In case that any WildFly instance crashes, clients know the location of the secondary broker and are able to fail over.
+--
+
+* Copy WildFly into two directories `wildfly-1` and `wildfly-2`
+* Start `wildfly-1` in `full-ha` profile in admin only mode and connect through CLI
+
+[source, bash ,options="nowrap"]
+----
+./wildfly-1/bin/standalone.sh -c standalone-full-ha.xml --admin-only
+----
+
+* In a different terminal, connect to the server using the JBoss CLI:
+
+[source, bash ,options="nowrap"]
+----
+./wildfly-1/bin/jboss-cli.sh -c
+----
+
+* Run following CLI commands on `wildfly-1`:
+
+[NOTE]
+--
+In the following CLI script change `<password>` and `<path>` values based on your environment.
+--
+
+[source, bash ,options="nowrap"]
+----
+# Change ActiveMQ Artemis cluster <password>
+/subsystem=messaging-activemq/server=default:write-attribute(name=cluster-password, value=<password>)
+
+# Set location of journal for pair A - change path to messaging-journal-a directory based on your environment
+/subsystem=messaging-activemq/server=default/path=bindings-directory:write-attribute(name=path,value=<path>/messaging-journal-a/bindings)
+/subsystem=messaging-activemq/server=default/path=journal-directory:write-attribute(name=path,value=<path>/messaging-journal-a/journal)
+/subsystem=messaging-activemq/server=default/path=large-messages-directory:write-attribute(name=path,value=<path>/messaging-journal-a/largemessages)
+/subsystem=messaging-activemq/server=default/path=paging-directory:write-attribute(name=path,value=<path>/messaging-journal-a/paging)
+
+# Set "default" Artemis broker as primary
+/subsystem=messaging-activemq/server=default/ha-policy=shared-store-primary:add(failover-on-server-shutdown=true)
+:reload
```

Comment:
s/:reload/reload/g

The string ':reload' should never appear in documentation. :) Users should always use the high-level command.